app/src/main/java/com/github/se/studybuddies/ui/video_call/VideoCallScreen.kt

```diff
@@ -1,44 +1,90 @@
 import io.getstream.video.android.compose.ui.components.call.activecall.CallContent
 import io.getstream.video.android.compose.ui.components.call.controls.ControlActions
-import io.getstream.video.android.compose.ui.components.call.controls.actions.CancelCallAction
 import io.getstream.video.android.compose.ui.components.call.controls.actions.FlipCameraAction
+import io.getstream.video.android.compose.ui.components.call.controls.actions.LeaveCallAction
 import io.getstream.video.android.compose.ui.components.call.controls.actions.ToggleCameraAction
 import io.getstream.video.android.compose.ui.components.call.controls.actions.ToggleMicrophoneAction
 import io.getstream.video.android.compose.ui.components.call.renderer.LayoutType
 import io.getstream.video.android.compose.ui.components.call.renderer.ParticipantVideo
 import io.getstream.video.android.compose.ui.components.call.renderer.ParticipantsLayout
 import io.getstream.video.android.compose.ui.components.call.renderer.RegularVideoRendererStyle
 import io.getstream.video.android.core.Call
+import io.getstream.video.android.core.RealtimeConnection
+import io.getstream.video.android.core.StreamVideo
 
 // Design UI elements using Jetpack Compose
 @Composable
-fun VideoCallScreen(call: Call, videoVM: VideoCallViewModel, onCallDisconnected: () -> Unit = {}) {
+fun VideoCallScreen(call: Call, onCallDisconnected: () -> Unit = {}) {
   val isCameraEnabled by call.camera.isEnabled.collectAsState()
   val isMicrophoneEnabled by call.microphone.isEnabled.collectAsState()
   val layout by remember { mutableStateOf(LayoutType.DYNAMIC) }
+  val connection by call.state.connection.collectAsStateWithLifecycle()
+  val context = LocalContext.current
 
   LockScreenOrientation(orientation = Configuration.ORIENTATION_PORTRAIT)
-  videoVM.joinCall()
+  LaunchedEffect(key1 = connection) {
+    when (connection) {
+      RealtimeConnection.Disconnected -> {
+        Toast.makeText(
+                context,
+                "Call connection disconnected (${(connection as RealtimeConnection.Failed).error}",
+                Toast.LENGTH_LONG,
+            )
+            .show()
+        onCallDisconnected.invoke()
+      }
+      RealtimeConnection.Failed("connection") -> {
+        Toast.makeText(
+                context,
+                "Call connection failed (${(connection as RealtimeConnection.Failed).error}",
+                Toast.LENGTH_LONG,
+            )
+            .show()
+        onCallDisconnected.invoke()
+      }
+      else -> {
+        if (StreamVideo.instance().state.activeCall.value == call) {
+          Log.d("MyPrint", "Active call is the same as the call we are trying to join")
+        } else {
+          try {
+            call.join()
+            Log.d("MyPrint", "Trying to join call")
+          } catch (e: Exception) {
+            Log.d("MyPrint", "Trying to join call got exception, leave call")
+            // call.leave()
+            // call.join()
```

Comment:
Could we display in the log what exception we good and remove the 2 commented lines (as we not really join the call, so we cannot leave it, and we don't want to join if there is something that didn't went well)

Reply:
It doesnt change anything.